Removed GitLab files, and updated some references to GitLab in our do…

kinow opened the pull request:
…cs and code.

Related to #1380

Was about to log off when I saw the .gitlab-ci.yml file in my IDE. Spent some 10 minutes~ searching and replacing some values in our docs/files.

The templates were deleted but we can easily grab the again by browsing the code in our latest tags in GitHub, then adjust the gitlab templates to github (for #2030).

isimo00 commented on docs/source/conf.py:
```diff
@@ -163,10 +163,9 @@
 }
 
 html_context = {
-    'gitlab_url': 'https://earth.bsc.es/gitlab',
```
Why no `github_url`?

kinow replied:
This is the way it's documented in the theme docs: https://github.com/pydata/pydata-sphinx-theme/blob/afc87beed0327d64083676d0fe355cd494a61f4c/docs/user_guide/source-buttons.rst#github

![image](https://github.com/user-attachments/assets/a1dc750e-4977-4e31-9094-ffa575118e39)

Looks like the generated GitHub URLs are correct, e.g. https://autosubmit--2127.org.readthedocs.build/en/2127/ (this PR generated page -- expand Checks below to find the link in the RTD check).

![image](https://github.com/user-attachments/assets/4d650077-4a5e-4df9-af62-c28317c9b5a6)

![image](https://github.com/user-attachments/assets/c01c9c8a-a41a-4a58-9c47-455aaa267205)

isimo00 replied:
Ah, great! Approved then :)

kinow replied:
Thanks Irene!

LuiggiTenorioK reviewed (commented):
- This is minor but we will probably need to change the GitLab icon on the home page.

![image](https://github.com/user-attachments/assets/47157ee8-2927-4ea1-8d7a-2405b79b93ab)

- Also, I saw that there is a GitLab issue referenced in the Installation page:

```diff
-.. warning:: This procedure is still WIP. You can follow the process at `issue #864 <https://earth.bsc.es/gitlab/es/autosubmit/-/issues/886>`_. We strongly recommend using the pip procedure.
+.. warning:: This procedure is still WIP. You can follow the process at `issue #864 <https://github.com/BSC-ES/autosubmit/issues/886>`_. We strongly recommend using the pip procedure.
```

- Then, I found some GitLab URLs in the EasyBuild recipe but not sure if that should be changed.

kinow commented:
@LuiggiTenorioK & @isimo00 I addressed your feedback (thanks for that!), could you please have another look. I haven't resolved the conversations so that it'd be easier for you both to review it.

LuiggiTenorioK reviewed (approved):
Looks good to me! :100:  Thanks @kinow!
